app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -35,7 +35,10 @@ def scsi_controller_type
       end
 
       def select_nic(fog_nics, nic)
-        fog_nics.detect {|fn| fn.network == nic.compute_attributes['network']} # grab any nic on the same network
+        nic_attrs = nic.compute_attributes
+        match =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
```

Comment:
@shlomizadok Not sure why you are getting back the portgroup as the foreman nic attribute name, if you do then the existing code should work for you, I get the non-portgroup name back:
compute_attributes: !ruby/hash:ActiveSupport::HashWithIndifferentAccess
type: VirtualVmxnet3
network: DPT_MAIN_PROD_200

So I need foreman to know the portgroup (dvportgroup-87 - the ID) which it currently doesn't know about:
irb(main):018:0> service.get_network('DPT_MAIN_PROD_200', datacenter)
=> {:id=>"dvportgroup-87", :name=>"DPT_MAIN_PROD_200", :accessible=>true, :datacenter=>"Turitea ITS"}

No problem changing the name for the match, I just used the same as was already in libvirt/server.rb